src/components/app.tsx

```diff
@@ -172,9 +172,10 @@ export class AppComponent extends BaseComponent<IProps, IState> {
       );
     }
 
+    const isPreviewing = urlParams.domain && urlParams.domain_uid && !urlParams.token;
     if (user.isStudent) {
       if (!groups.groupForUser(user.id)) {
-        if (appConfig.autoAssignStudentsToIndividualGroups) {
+        if (appConfig.autoAssignStudentsToIndividualGroups || isPreviewing) {
```

Comment:
Can't we use `this.stores.isPreviewing` here?
```suggestion
        if (appConfig.autoAssignStudentsToIndividualGroups || this.stores.isPreviewing) {
```
That way we don't have redundant code for detecting preview and we also don't need to figure it out on every render.

Reply:
At present in my changes `isPreviewing` is only an argument to the `createStores` function and isn't kept in a store for future use.  We could. however, add it there.  Seems worth doing.

Reply:
I added a commit that does the above and reduces the redundant code.